app/services/CheckoutService.scala

```diff
@@ -70,8 +70,12 @@ class CheckoutService(identityService: IdentityService,
 
     val updatedCommand = additionalRateplan.map { ratePlantoAdd =>
       val updatedRatePlans = ratePlantoAdd <:: originalCommand.ratePlans
-      val updatedContractEffective = DateTime.now.toLocalDate.plusDays(defaultPaymentDelayt.getDays)
-      originalCommand.copy(ratePlans = updatedRatePlans, contractEffective = updatedContractEffective, contractAcceptance = updatedContractEffective.plusDays(42))
+
+      val defaultDelayDays = defaultPaymentDelay.getDays
+      val introductoryPeriodDelayDays = if (defaultDelayDays > 0) defaultDelayDays - 1  else defaultDelayDays
+
+      val updatedContractEffective = DateTime.now.toLocalDate.plusDays(introductoryPeriodDelayDays)
+      originalCommand.copy(ratePlans = updatedRatePlans, contractEffective = updatedContractEffective, contractAcceptance = updatedContractEffective.plusDays(43))
```

Comment:
👎 this will break for 12-for-12. Is there somewhere in the model you can get the 6 or 12 number from? Then you can do something like:
```
.plusWeeks(sub.plan.product.introductoryPeriodWeeks).plusDays(1)
```